alfresco-health-processor-platform/src/main/java/eu/xenit/alfresco/healthprocessor/reporter/AttributeHealthReportsStore.java

```diff
@@ -0,0 +1,89 @@
+package eu.xenit.alfresco.healthprocessor.reporter;
+
+import eu.xenit.alfresco.healthprocessor.plugins.api.HealthProcessorPlugin;
+import eu.xenit.alfresco.healthprocessor.reporter.api.NodeHealthReport;
+import eu.xenit.alfresco.healthprocessor.reporter.api.NodeHealthStatus;
+import eu.xenit.alfresco.healthprocessor.util.AttributeStore;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import org.alfresco.util.Pair;
+
+/**
+ * {@link HealthReportsStore} implementation that persists stats and failing reports via the {@link
+ * org.alfresco.service.cmr.attributes.AttributeService}.
+ * </p>
+ * Reports and stats are stored using the following key value combination:
+ * <ul>
+ *     <li>health-processor.reports.${random-uuid}=${Pair(PluginClass, NodeHealthReport)}</li>
+ *     <li>health-processor.report-stats.${PluginClass}=${HashMap(NodeHealthStatus, Count)}</li>
+ * </ul>
+ */
+@RequiredArgsConstructor
+public class AttributeHealthReportsStore implements HealthReportsStore {
+
+    private static final String ATTR_KEY_REPORTS = "reports";
+    private static final String ATTR_KEY_REPORT_STATS = "report-stats";
+
+    private final AttributeStore attributeStore;
+
+    @Override
+    public void storeUnhealthyReport(Class<? extends HealthProcessorPlugin> pluginClass, NodeHealthReport report) {
+        Pair<Class<? extends HealthProcessorPlugin>, NodeHealthReport> pluginClassWithReport =
+                new Pair<>(pluginClass, report);
+        attributeStore.setAttribute(pluginClassWithReport, ATTR_KEY_REPORTS, UUID.randomUUID().toString());
+    }
+
+    @Override
+    public void recordReportStats(Class<? extends HealthProcessorPlugin> pluginClass, Set<NodeHealthReport> reports) {
+        HashMap<NodeHealthStatus, Long> oldStats = attributeStore.getAttribute(ATTR_KEY_REPORT_STATS, pluginClass);
+        if (oldStats == null) {
+            oldStats = new HashMap<>();
```

Comment:
SonarCloud: Use EnumMap here. EnumMap has a more efficient internal representation.

Reply:
Using an `EnumMap` results in following exception:
```
2021-04-19 07:05:40,718  ERROR [quartz.core.ErrorLogger] [DefaultScheduler_Worker-1] Job (DEFAULT.eu.xenit.health-processor threw an exception.
org.quartz.SchedulerException: Job threw an unhandled exception. [See nested exception: org.springframework.scheduling.quartz.JobMethodInvocationFailedException: Invocation of method 'startIfNotRunning' on target class [class eu.xenit.alfresco.healthprocessor.processing.ProcessorTask] failed; nested exception is java.lang.ClassCastException: class java.util.HashMap cannot be cast to class java.util.EnumMap (java.util.HashMap and java.util.EnumMap are in module java.base of loader 'bootstrap')]
	at org.quartz.core.JobRunShell.run(JobRunShell.java:213)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
Caused by: org.springframework.scheduling.quartz.JobMethodInvocationFailedException: Invocation of method 'startIfNotRunning' on target class [class eu.xenit.alfresco.healthprocessor.processing.ProcessorTask] failed; nested exception is java.lang.ClassCastException: class java.util.HashMap cannot be cast to class java.util.EnumMap (java.util.HashMap and java.util.EnumMap are in module java.base of loader 'bootstrap')
	at org.springframework.scheduling.quartz.MethodInvokingJobDetailFactoryBean$MethodInvokingJob.executeInternal(MethodInvokingJobDetailFactoryBean.java:276)
	at org.springframework.scheduling.quartz.QuartzJobBean.execute(QuartzJobBean.java:75)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
	... 1 more
Caused by: java.lang.ClassCastException: class java.util.HashMap cannot be cast to class java.util.EnumMap (java.util.HashMap and java.util.EnumMap are in module java.base of loader 'bootstrap')
	at eu.xenit.alfresco.healthprocessor.reporter.AttributeHealthReportsStore.recordReportStats(AttributeHealthReportsStore.java:45)
```

When the cycle picks-up it's old state after restarting Alfresco...

Reply:
Remains unclear to me why this would be a HashMap, but I approved.

Reply:
I changed to `EnumMap` in the last commit. 

I tried to debug this further, however I cannot pinpoint the exact where Alfresco makes a `HashMap` of the value when retrieving the attribute from the DB. 
However the behavior is clear:

- when Alfresco gets the attribute from cache, the `EnumMap` type is respected
- when Alfresco performs an entity lookup to fetch the attribute, the value is of type `HashMap`